File: easy/backspace_compare.py
def process_backspace_compare(text: str) -> str:
    result = []
    for c in text:
        if c == "#":
            if len(result) != 0:
                result.pop(-1)
        else:
            result.append(c)
    return "".join(result)


def backspace_compare(s: str, t: str) -> str:
    return process_backspace_compare(s) == process_backspace_compare(t)

File: lib/easy/backspace_compare.dart
String processBackspaceCompare(String text) {
  List<int> result = [];
  for (int char in text.codeUnits) {
    if (char == 35) {
      if (result.isNotEmpty) {
        result.removeLast();
      }
    } else {
      result.add(char);
    }
  }
  return result.toString();
}

bool backspaceCompare(String s, String t) {
  return processBackspaceCompare(s) == processBackspaceCompare(t);
}
